src/main/java/org/apache/datasketches/memory/UnsafeUtil.java

```diff
@@ -157,6 +157,30 @@ static long getFieldOffset(final Class<?> c, final String fieldName) {
     }
   }
 
+  static long getArrayBaseOffset(final Class<?> c) {
+    if (c == byte[].class) {
```

Comment:
I suggest using a static `IdentityHashMap` instead.

Reply:
Why - do you think it will be faster? I haven't experimented, but I could believe that a relatively short if/else chain with reference equality checks would beat a hashtable.